designs/instance-type-settings.md

```diff
@@ -0,0 +1,340 @@
+
+1. There is still [users asks](https://github.com/aws/karpenter/issues/1803) for calculating the kubeReserved resources value based on the GKE calculation of the kubeReserved resources value as opposed to the EKS default which is based on pod density. Should we allow some flag on the provisioner that can determine which way to base the calculation on? i.e. `kubeReservedMemoryStrategy`
+
+### Extending Ephemeral Storage
+
+There are [user asks](https://github.com/aws/karpenter/issues/1995) to have the ephemeral storage dynamically scaled for nodes based on the pod requests that are scheduled to that node (combined with overhead for kubeletReserved and systemReserved). Ephemeral storage is tied to blockDeviceMappings specified in launch templates. We currently specify default ephemeral storage of 20GB for all AMI families unless the user specifies their own custom launch template or explicitly specifies their own custom blockDeviceMappings.
+
+In general, it seems the asks from users specify the ephemeral-storage as a function of pod density on the node. We could consider adding `blockDeviceMappings` to the `InstanceType` CR; however, this places more maintenance overhead on the user and makes the `InstanceType` CR vendor-specific.
+
+`InstanceType` CR with Block Device Mappings:
+
+```yaml
+apiVersion: karpenter.sh/v1alpha1
+kind: InstanceType
+metadata:
+  name: "c5.large"
+spec:
+  allocatable: # This defines the starting resources that the instance has
+    custom-device.com/resource: "2"
+    memory: 1Gi
+    cpu: 2
+  provider:
+    blockDeviceMappings:
+    - deviceName: /dev/xvda
+      ebs:
+        volumeSize: 100Gi
+        volumeType: gp3
+        iops: 10000
+        encrypted: true
+        kmsKeyID: "arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab"
+        deleteOnTermination: true
+        throughput: 125
+        snapshotID: snap-0123456789
+```
+
+
+**🔑 Recommendation:** Based on the additional overhead and muddiness of putting `blockDeviceMappings` in the `InstanceType` CR, we will punt on block device mappings for now. For now, we will consider extending ephermal-storage through some dynamic mechanism out of scope, though the hope is that we will have some mechanism as a future feature that calculates the combined needed ephemeral-storage based off of the pod requests, kubeReserved and systemReserved.
+
+### Launch Templates
+
+For now, we are avoiding the need for Instance Type changes to affect launch template cardinality. Configuration at the `InstanceType` level is intended to configure the starting point for the scheduler to know how many resources can be used by the bin-packer. The expectation is that when the node comes up this is roughly the allocatable amount of resources the node will have, less kubeReserved, systemReserved, and evictionThresholds. Thus, the calculation we will do will be:
+
+```
+Allocatable = Instance Type Resources - kubeReserved - systemReserved - evictionThreshhold
```

Comment:
Now that I see this, my recommendation elsewhere to have the field be `InstanceType.Spec.Allocatable` is inaccurate. Is it more correct to make it `InstanceType.Spec.Capacity` and then we substract off these additional 3 values that are controlled at the provisioner level?

Reply:
Yes, you are completely right. This should be `Capacity` not `Allocatable`